source/mesh_refinement/composition_gradient.cc

```diff
@@ -19,6 +19,7 @@
 */
 
 
+#include <aspect/simulator.h>
```

Comment:
Ah, we really need to find a solution to put `AdvectionField` out of the `simulator.h` header. Including `simulator.h` in a lot of places is not a clean design. Not for this PR though.

Reply:
agreed, see #5887